Projects/Modules/NetworkModule/Sources/Client/NetworkClient.swift

```diff
@@ -32,16 +34,35 @@ private extension NetworkClient {
     }
     
     func requestNetworkTask(with request: URLRequest, from endpoint: E) async throws -> Response {
-        let (data, response) = try await session.data(for: request)
+        let (data, urlResponse) = try await session.data(for: request)
+        let response = Response(request: request, data: data, response: urlResponse)
+        try interceptResponse(with: response, from: endpoint)
         
-        guard let httpResponse = response as? HTTPURLResponse else { throw NetworkError.invaildResponse }
+        guard let httpResponse = urlResponse as? HTTPURLResponse else { throw NetworkError.invaildResponse }
         
         let statusceCode = httpResponse.statusCode
         
         if !(endpoint.validationCode ~= statusceCode) {
             throw HTTPError(statuscode: statusceCode)
         }
     
-        return Response(request: request, data: data, response: response)
+        return response
+    }
+    
+    func interceptRequest(with request: URLRequest, from endpoint: E) throws -> URLRequest {
+        var requst = request
+        
+        for interceptor in self.interceptors {
+            try interceptor.willRequest(requst, from: endpoint)
+            requst = try interceptor.prepare(requst, from: endpoint)
+        }
+        
+        return requst
+    }
+    
+    func interceptResponse(with response: Response, from endpoint: E) throws {
+        for interceptor in self.interceptors {
+            try interceptor.didRecieve(response, from: endpoint)
+        }
     }
```

Comment:
아래와 같이 적극적으로 고차함수를 활용해보는것도 좋아보입니다. 물론.. 개인적인... 

```suggestion
    func interceptRequest(with request: URLRequest, from endpoint: E) throws -> URLRequest {
        try interceptors.reduce(request) { currentRequest, interceptor in
            try interceptor.willRequest(currentRequest, from: endpoint)
            return try interceptor.prepare(currentRequest, from: endpoint)
        }
    }
    
    func interceptResponse(with response: Response, from endpoint: E) throws {
        try interceptors.forEach { try $0.didRecieve(response, from: endpoint) }
    }
```

Reply:
오 많이 배워갑니다 ! 적용할께요